src/tribler/core/components/ipv8/eva/protocol.py

```diff
@@ -91,17 +117,20 @@ def __init__(  # pylint: disable=too-many-arguments
         """
         self.community = community
 
-        self.scheduled = defaultdict(deque)
         self.settings = settings or EVASettings()
 
-        self.on_send_complete = on_send_complete
-        self.on_receive = on_receive
-        self.on_error = on_error
+        self.on_send_complete = on_send_complete or blank
+        self.on_receive = on_receive or blank
+        self.on_error = on_error or blank
+        self.on_request = on_request
 
-        self.incoming: Dict[Peer, IncomingTransfer] = {}
-        self.outgoing: Dict[Peer, OutgoingTransfer] = {}
+        self.incoming = Container(self)
+        self.outgoing = Container(self)
```

Comment:
The information about the container item type was lost. It should be here for PyCharm to understand the type of `transfer.window` in `on_data_packet`, etc.:
```suggestion
        self.incoming: Container[Peer, IncomingTransfer] = Container(self)
        self.outgoing: Container[Peer, OutgoingTransfer] = Container(self)
```